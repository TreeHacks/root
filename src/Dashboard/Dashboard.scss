@import "../Global.scss";
.dashboard {
    width: 100%;
    height: calc(100vh - 80px);
    background-repeat: no-repeat;
    background-size: 100% 100%;

    .treehacks-dashboard-message-container {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translateX(-50%) translateY(-50%);
        max-height: calc(100vh - 200px);
    }

    .days-left {
        color: $primary-color;
        font-size: 70px;
        margin-bottom: -15px;
    }

	.dashboard-design {
		color: $text-color;
        text-align: center;
        background-color: $background-color;
        width: 100%;
        max-width: 500px;
        min-width: 300px;
        padding: 20px;
        border-radius: 5px;
        border: 2px solid $primary-color;
        transition: 0.2s;

        & + .dashboard-design {
            margin-top: 20px;
        }

        &.notice {
            padding: 1rem 1rem 0.8rem;
            margin-top: 5rem;

            a {
                color: $primary-color;
            }
        }

        &:hover {
            transform: scale(1.02);
        }
	}

    .admitted-content {
        padding-top: 1rem;

        .btn {
            margin-bottom: 1rem;
        }

        a {
            color: $primary-color;
        }
    }
}

@media only screen and (max-width: 1024px) {
    .dashboard {
        background-size: 200vh;
    }
}

/*.dashboard-body {
    background-image: url(/art/circuit_left.svg), url(/art/circuit_bottom.svg);
    background-position: left top, right bottom;
    background-repeat: no-repeat, no-repeat;
    height: 100%;
}*/
