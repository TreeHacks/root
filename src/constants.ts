import settings from "./themes/settings";

export var GROUPS = {
  'admin': 'Administrator',
  'reviewer': 'Reviewer',
  'sponsor': 'Sponsor',
  'judge': 'Judge'
};

export var STATUS = {
  INCOMPLETE: "incomplete",
  SUBMITTED: "submitted",
  WAITLISTED: "waitlisted",
  REJECTED: "rejected",
  ADMITTED: "admitted",
  ADMISSION_CONFIRMED: "admission_confirmed",
  ADMISSION_DECLINED: "admission_declined"
}

export var TYPE = {
  IN_STATE: "is",
  OUT_OF_STATE: "oos",
  STANFORD: "stanford"
}

export var DEADLINES = settings.deadlines;

export var TRANSPORTATION_STATUS = {
  UNAVAILABLE: "unavailable",
  AVAILABLE: "available",
  SUBMITTED: "submitted",
  REJECTED: "rejected",
  APPROVED: "approved",
  PAID: "paid"
};

export var TRANSPORTATION_TYPES = {
  BUS: "bus",
  FLIGHT: "flight",
  OTHER: "other"
};

export var TRANSPORTATION_BUS_ROUTES = {
  TEST: "test",
  TEST_NO_COORDINATOR: "test_no_coordinator",
  USC: "usc",
  UCLA: "ucla",
  SANDIEGO: "sandiego",
  UCI: "uci",
  POMONA: "pomona",
  BERKELEY: "berkeley"
};

export var TRANSPORTATION_DEADLINES = {
  [TRANSPORTATION_TYPES.FLIGHT]: 'December 9th at 11:59pm PST',
  [TRANSPORTATION_TYPES.BUS]: 'January 10th at 11:59pm PST',
  [TRANSPORTATION_TYPES.OTHER]: 'February 19th at 11:59pm PST'
};


export type IBusRoute = {
  day?: string,
  time?: string,
  stop?: string,
  location?: string,
  hack?: boolean
};

export var TRANSPORTATION_BUS_ROUTE_DETAILS: { [x: string]: { coordinator: { name: string, email: string }, route: IBusRoute[] } } = {
  [TRANSPORTATION_BUS_ROUTES.USC]: {
    coordinator: { 'name': 'Caleb Thomas', 'email': 'calebtho@usc.edu' },
    route: [
      {
        day: 'Friday, February 15th',
        time: '8:30am',
        stop: 'Check in at 37th & McClintock',
        location: 'Los Angeles'
      },
      {
        day: 'Friday, February 15th',
        time: '4:30pm',
        stop: 'Panama St. & Via Ortega',
        location: 'Stanford'
      },
      {
        hack: true
      },
      {
        day: 'Sunday, February 17th',
        time: '3:30pm',
        stop: 'Check in at Panama St. & Via Ortega',
        location: 'Stanford'
      },
      {
        day: 'Sunday, February 17th',
        time: '10:30pm',
        stop: '37th & McClintock',
        location: 'Los Angeles'
      }
    ]
  },
  [TRANSPORTATION_BUS_ROUTES.TEST]: {
    coordinator: { 'name': 'Tree Hack', 'email': 'treehack@treehacks.com' },
    route: [
      {
        day: 'Friday, February 15th',
        time: '8:00am',
        stop: 'Check in at 37th & McClintock',
        location: 'Atlanta'
      },
      {
        hack: true
      },
      {
        day: 'Sunday, February 17th',
        time: '11:00pm',
        stop: '37th & McClintock',
        location: 'Los Angeles'
      }
    ]
  },
  [TRANSPORTATION_BUS_ROUTES.TEST_NO_COORDINATOR]: {
    coordinator: null,
    route: [
      {
        day: 'Friday, February 15th',
        time: '8:00am',
        stop: 'Check in at 37th & McClintock',
        location: 'Atlanta'
      },
      {
        hack: true
      },
      {
        day: 'Sunday, February 17th',
        time: '11:00pm',
        stop: '37th & McClintock',
        location: 'Los Angeles'
      }
    ]
  },
  [TRANSPORTATION_BUS_ROUTES.UCLA]: {
    coordinator: { 'name': 'Meera Rachamallu', 'email': 'mrachamallu@g.ucla.edu' },
    route: [
      {
        day: 'Friday, February 15th',
        time: '9:00am',
        stop: 'Check in at 10 Charles E. Young N',
        location: 'Los Angeles'
      },

      {
        day: 'Friday, February 15th',
        time: '4:30pm',
        stop: 'Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        hack: true
      },

      {
        day: 'Sunday, February 17th',
        time: '3:30pm',
        stop: 'Check in at Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        day: 'Sunday, February 17th',
        time: '10:30pm',
        stop: '10 Charles E. Young N',
        location: 'Los Angeles'
      }
    ]
  },
  [TRANSPORTATION_BUS_ROUTES.SANDIEGO]: {
    coordinator: null,
    route: [
      {
        day: 'Friday, February 15th',
        time: '7:30am',
        stop: 'Check in at UCSD (location added later)',
        location: 'Los Angeles'
      },

      {
        day: 'Friday, February 15th',
        time: '6:00pm',
        stop: 'Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        hack: true
      },

      {
        day: 'Sunday, February 17th',
        time: '3:30pm',
        stop: 'Check in at Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        day: 'Sunday, February 17th',
        time: '11:45pm',
        stop: 'UCSD',
        location: 'San Diego'
      }
    ]
  },
  [TRANSPORTATION_BUS_ROUTES.UCI]: {
    coordinator: { 'name': 'Tara Porter', 'email': 'tporter@caltech.edu' },
    route: [
      {
        day: 'Friday, Februrary 15th',
        time: '7:30am',
        stop: 'Check-in in front of Avery House (293 S Holliston Ave, Pasadena, CA)',
        location: 'Pasadena'
      },

      {
        day: 'Friday, Feburary 15th',
        time: '4:30pm',
        stop: 'Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        hack: true
      },

      {
        day: 'Sunday, Februrary 17th',
        time: '3:30pm',
        stop: 'Check in at Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        day: 'Sunday, Februrary 17th',
        time: '11:00pm',
        stop: 'In front of Avery House (293 S Holliston Ave, Pasadena, CA)',
        location: 'Pasadena'
      }
    ]
  },
  [TRANSPORTATION_BUS_ROUTES.POMONA]: {
    coordinator: { 'name': 'Meera Rachamallu', 'email': 'mrachamallu@g.ucla.edu' },
    route: [
      {
        day: 'Friday, Februrary 15th',
        time: '10:45am',
        stop: 'Check in at Harvey Mudd (location added later)',
        location: 'Claremont'
      },

      {
        day: 'Friday, Feburary 15th',
        time: '6:00pm',
        stop: 'Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        hack: true
      },

      {
        day: 'Sunday, Februrary 17th',
        time: '3:30pm',
        stop: 'Check in at Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        day: 'Sunday, Februrary 17th',
        time: '10:00pm',
        stop: 'Harvey Mudd',
        location: 'Claremont'
      }
    ]
  },
  [TRANSPORTATION_BUS_ROUTES.BERKELEY]: {
    coordinator: { 'name': 'Adhiv Dhar', 'email': 'adhiv@calhacks.io' },
    route: [
      {
        day: 'Friday, Februrary 15th',
        time: '2:30pm',
        stop: 'Check in at West Circle / University Drive',
        location: 'Berkeley'
      },

      {
        day: 'Friday, Feburary 15th',
        time: '4:30pm',
        stop: 'Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        hack: true
      },

      {
        day: 'Sunday, Februrary 17th',
        time: '3:30pm',
        stop: 'Check in at Panama St. & Via Ortega',
        location: 'Stanford'
      },

      {
        day: 'Sunday, Februrary 17th',
        time: '5:30pm',
        stop: 'West Circle / University Drive',
        location: 'Berkeley'
      }
    ]
  }
};

export const HACKATHON_YEAR = settings.hackathon_year;
export const HACKATHON_DATE_RANGE = settings.hackathon_date_range;
export const LOCATIONS = settings.locations;
export const applicationReviewDisplayFields = ["first_name", "last_name", "university", "graduation_year", "level_of_study", "major", "skill_level", "hackathon_experience", "resume", "q1_goodfit", "q2_experience", "q3", "q4"];
export const sponsorApplicationDisplayFields = ["first_name", "last_name", "university", "graduation_year", "level_of_study", "major", "resume", "q2_experience", "q4"];
export const stanfordApplicationDisplayFields = [
  "first_name",
  "last_name",
  "phone",
  "dob",
  "gender",
  "race",
  "university",
  "graduation_year",
  "level_of_study",
  "major",
  "skill_level",
  "hackathon_experience",
  "resume",
  "section2",
  // "q1_goodfit",
  // "q2_experience",
  // "q3",
  "q4",
  "accept_terms",
  "accept_share"
];

export const VERTICALS = ["health", "safety", "awareness"];

export const FLOORS = [0, 1, 2, 3];