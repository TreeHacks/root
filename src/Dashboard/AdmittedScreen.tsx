import React from "react";
import { connect } from "react-redux";
import { HACKATHON_YEAR, HACKATHON_DATE_RANGE } from "../constants";
import { confirmAdmission, declineAdmission } from "../store/form/actions";
import { Link } from "react-router-dom";
import moment from "moment-timezone";

interface IAdmittedScreenProps {
  confirmAdmission: () => void;
  declineAdmission: () => void;
  confirmedYet: boolean;
  deadline: string;
}
export const AdmittedScreen = (props: IAdmittedScreenProps) => (
  <div className="admitted-content">
    {props.confirmedYet && (
      <div>
        <h4>You're coming to TreeHacks {HACKATHON_YEAR}!</h4>
        <p>Admission confirmed &ndash; see you there!</p>
        <p>Complete the following by <strong>January 24th:</strong> </p>
        <ul>
          <li>Follow the steps in your <Link to="/transportation">travel section</Link> to confirm your travel plans (we have very specific instructions on travelling to TreeHacks, please read carefully)! You must also input a mailing address that will be valid and accessible by you till May 2024 at the end of that page. </li>
          <li>Register through GrantEd: complete this form to register as a vendor (this is required in order to receive a prize or a travel reimbursement)</li>
          <li>Sign the <a href=""> liability waiver </a> (need to print and then sign, virtual signature not accepted) and <a href=""> upload it here </a> </li>
        </ul>
        <p>
          Keep an eye on your inbox in the coming weeks -- we'll be sending more
          details to you shortly!
        </p>
        <button className="btn btn-custom">
          <Link to="/transportation" style={{ color: "white" }}>
            Travel RSVP details
          </Link>
        </button>
        <p>
          Check out {" "}
          <a href="https://docs.google.com/document/d/162lC3yzvkURS1FZcVX-vpMckuWYkbqnoESZcPdxRdHk/edit?usp=sharing">
            Ultimate TreeHacks Guide ðŸŒ²
          </a>{" "}
          to learn more!
        </p>
      </div>
    )}
    {!props.confirmedYet && (
      <div>
        <h4>
          Congratulations! You've been accepted to TreeHacks {HACKATHON_YEAR}!
        </h4>
        <p>
          We're so stoked to see you the weekend of {HACKATHON_DATE_RANGE} at
          TreeHacks {HACKATHON_YEAR}! We were blown away by your application and
          know that youâ€™ll be an inspiring hacker. It'll be a super fun weekend
          filled with amazing hacks, and fantastic people. We hope you can join
          us!
        </p>
        <p>
          You have until{" "}
          {moment(props.deadline)
            .tz("America/Los_Angeles")
            .format("LLL z")}{" "}
          to confirm your attendance at TreeHacks {HACKATHON_YEAR}. If you don't
          confirm by then, we'll assume you can't make it and give your spot to
          someone else.
        </p>
        <p>
          By confirming your spot, you agree to let us share the information you
          gave us in your application with our sponsors. If you'd like to opt
          out, send us a message at hello@treehacks.com with the subject "Sponsor Information Opt Out".
        </p>
      </div>
    )}
    <button
      className="btn btn-custom inverted"
      onClick={() =>
        window.confirm(
          `Are you sure you would like to decline your spot at TreeHacks ${HACKATHON_YEAR}? Your decision is final.`
        ) && props.declineAdmission()
      }
    >
      decline spot
    </button>
    <button
      className="btn btn-custom"
      disabled={props.confirmedYet}
      onClick={() => props.confirmAdmission()}
    >
      {props.confirmedYet ? "already confirmed" : "confirm spot"}
    </button>
    {!props.confirmedYet && (
      <>
        <p>
          We are committed to helping every admitted hacker get here! Check out
          the <Link to="/transportation">travel section</Link> for details.
        </p>
      </>
    )}
  </div>
);

const mapStateToProps = (state) => ({});

const mapDispatchToProps = (dispatch, ownProps) => ({
  confirmAdmission: () => dispatch(confirmAdmission()),
  declineAdmission: () => dispatch(declineAdmission()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdmittedScreen);
