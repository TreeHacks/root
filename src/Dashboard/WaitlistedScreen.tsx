import React from "react";
import { HACKATHON_YEAR } from "../constants";

export default () => (
  <div className="admitted-content">
    <h4>You are on the waitlist for TreeHacks {HACKATHON_YEAR}.</h4>
    <p>
      We were really impressed by your application, but we unfortunately cannot
      offer you a spot at this time.
    </p>
    <p>
      We will contact you by February 1st, 2025 with more information if we're
      able to offer you a spot at TreeHacks. Stay tuned and be on the lookout
      for updates from us in the future.
    </p>
    <p>Best,</p>
    <p>The TreeHacks Team</p>
  </div>
);
