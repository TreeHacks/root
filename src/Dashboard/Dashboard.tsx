import React from "react";
import { connect } from "react-redux";
import { getUserProfile } from "../store/form/actions";
import { IFormState } from "../store/form/types";
import { IDashboardProps, IDashboardWrapperProps } from "./types";
import Loading from "../Loading/Loading";
import { DEADLINES, STATUS } from '../constants';
import "./Dashboard.scss";
import AdmittedScreen from "./AdmittedScreen";
import AdmissionExpiredScreen from "./AdmissionExpiredScreen";
import AdmissionDeclinedScreen from "./AdmissionDeclinedScreen";
import { get } from "lodash";
import WaitlistedScreen from "./WaitlistedScreen";
import RejectedScreen from "./RejectedScreen";
import moment from "moment-timezone";
import { Link } from "react-router-dom";

export const Dashboard = (props: IDashboardProps) => {
    const deadline = DEADLINES.find(d => d.key === (props.profile.type || 'oos'));
    const deadlineDate = moment(deadline.date);
    const dateNow = moment();
    // const diffDays = deadlineDate.diff(dateNow, "days");
    const diffDays = Math.round(Math.abs((deadlineDate.valueOf() - dateNow.valueOf()) / (24 * 60 * 60 * 1000)));
    const displayDeadline = deadline.display_date || deadlineDate.toLocaleString('en-US', { month: 'long', year: 'numeric', day: 'numeric' });
    const acceptanceConfirmDeadline = get(props.profile, "admin_info.acceptance.deadline");
    const acceptanceConfirmDeadlineObject = moment(acceptanceConfirmDeadline);
    return (
        <div className="dashboard" style={{ "backgroundImage": `url('${require('../art/combined_circuit.svg')}')` }}>
            <div className="treehacks-dashboard-message-container">
                {props.profile.status === STATUS.ADMISSION_CONFIRMED ?
                    <div className="dashboard-design notice">
                        Time to hack! Looking to <Link to="/rooms">reserve a room</Link> for your team?
                    </div>
                : null}
                <div className="dashboard-design">
                    {
                        props.profile.status === STATUS.REJECTED ? <RejectedScreen /> :
                        props.profile.status === STATUS.WAITLISTED ? <WaitlistedScreen /> :
                        props.profile.status === STATUS.ADMISSION_CONFIRMED ? <AdmittedScreen confirmedYet={true} /> :
                        props.profile.status === STATUS.ADMISSION_DECLINED ? <AdmissionDeclinedScreen /> :
                        props.profile.status === STATUS.ADMITTED && dateNow > acceptanceConfirmDeadlineObject ? <AdmissionExpiredScreen /> :
                        props.profile.status === STATUS.ADMITTED ? <AdmittedScreen confirmedYet={false} deadline={acceptanceConfirmDeadline} /> :
                        props.profile.status === STATUS.SUBMITTED ? (
                                <span>
                                    Your application has been received &ndash; you are all good for now!<br /><br />We will email you when decisions are released and will handle any travel questions at that time. Thanks for applying :)
                                </span>
                            ) : dateNow > deadlineDate ? (
                                <span>Sorry, the application window has closed.</span>
                            ) : (
                                        <div>
                                            <div>
                                                You haven't submitted your application yet. You have
                                            </div>
                                            <div className="days-left">
                                                {diffDays}
                                            </div>
                                            <div>
                                                days to submit your application before the deadline:<br /><strong>{displayDeadline}</strong>.
                                </div>
                                        </div>
                                    )
                    }
                </div>
            </div>
        </div>
    );
}

const mapStateToProps = state => ({
    ...state.form
});

const mapDispatchToProps = (dispatch, ownProps) => ({
    getUserProfile: () => dispatch(getUserProfile())
});

class DashboardWrapper extends React.Component<IDashboardWrapperProps, {}> {
    componentDidMount() {
        this.props.getUserProfile();
    }
    render() {
        if (!this.props.profile) {
            return <Loading />;
        }
        return <Dashboard profile={this.props.profile} />;
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(DashboardWrapper);
