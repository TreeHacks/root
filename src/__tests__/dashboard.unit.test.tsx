import React from "react";
import { shallow, mount, render } from 'enzyme';
import { Provider } from "react-redux";
import store from "../store";
import sinon from "sinon";
import lolex from "lolex";
import {Dashboard} from "../Dashboard/Dashboard";
import { STATUS, TYPE, DEADLINES } from "../constants";
import AdmittedScreen from "../Dashboard/AdmittedScreen";
import AdmissionDeclinedScreen from "../Dashboard/AdmissionDeclinedScreen";
import AdmissionExpiredScreen from "../Dashboard/AdmissionExpiredScreen";
import RejectedScreen from "../Dashboard/RejectedScreen";
import WaitlistedScreen from "../Dashboard/WaitlistedScreen";

it('dashboard incomplete after deadline', () => {
    const profile = {
        status: STATUS.INCOMPLETE,
        type: TYPE.OUT_OF_STATE
    };
    const clock = lolex.install({now: new Date("01/01/2048")});
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    clock.uninstall();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toContain("Sorry, the application window has closed.");
});

it('dashboard incomplete before deadline', () => {
    const profile = {
        status: STATUS.INCOMPLETE,
        type: TYPE.OUT_OF_STATE
    };
    const clock = lolex.install({now: new Date("01/01/1999")});
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    clock.uninstall();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toContain("You haven't submitted your application yet.");
    expect(wrapper.text()).toContain("before the deadline:November 19,");
});

it('dashboard submitted', () => {
    const profile = {
        status: STATUS.SUBMITTED,
        type: TYPE.OUT_OF_STATE
    };
  
    const wrapper = render(
        <Dashboard profile={ profile } />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toContain("Your application has been received");
});

it('dashboard admission confirmed', () => {
    const profile = {
        status: STATUS.ADMISSION_CONFIRMED,
        type: TYPE.OUT_OF_STATE
    };
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(<AdmittedScreen confirmedYet={true} />)).toBe(true);
});

it('dashboard admission declined', () => {
    const profile = {
        status: STATUS.ADMISSION_DECLINED,
        type: TYPE.OUT_OF_STATE
    };
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(<AdmissionDeclinedScreen />)).toBe(true);
});

it('dashboard admitted', () => {
    const DEADLINE = "2048-01-30T04:39:47.512Z";
    const profile = {
        status: STATUS.ADMITTED,
        type: TYPE.OUT_OF_STATE,
        admin_info: {
            acceptance: {
                deadline: DEADLINE
            }
        }
    };
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(<AdmittedScreen confirmedYet={false} deadline={DEADLINE} />)).toBe(true);
});

it('dashboard admission confirm expired', () => {
    const profile = {
        status: STATUS.ADMITTED,
        type: TYPE.OUT_OF_STATE,
        admin_info: {
            acceptance: {
                deadline: "2018-01-30T04:39:47.512Z"
            }
        }
    };
    const clock = lolex.install({now: new Date("2018-01-30T04:39:48.512Z") });
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    clock.uninstall();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(<AdmissionExpiredScreen />)).toBe(true);
});

it('dashboard rejected', () => {
    const profile = {
        status: STATUS.REJECTED,
        type: TYPE.OUT_OF_STATE,
        admin_info: {}
    };
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(<RejectedScreen />)).toBe(true);
});

it('dashboard waitlisted', () => {
    const profile = {
        status: STATUS.WAITLISTED,
        type: TYPE.OUT_OF_STATE,
        admin_info: {}
    };
  
    const wrapper = shallow(
        <Dashboard profile={ profile } />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(<WaitlistedScreen />)).toBe(true);
});