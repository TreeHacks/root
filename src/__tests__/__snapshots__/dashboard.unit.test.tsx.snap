// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard admission confirm expired 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design"
    >
      <Component />
    </div>
  </div>
</div>
`;

exports[`dashboard admission confirmed 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design notice"
    >
      Time to hack! Looking to 
      <Link
        replace={false}
        to="/rooms"
      >
        reserve a room
      </Link>
       for your team?
    </div>
    <div
      className="dashboard-design"
    >
      <Connect(Component)
        confirmedYet={true}
      />
    </div>
  </div>
</div>
`;

exports[`dashboard admission declined 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design"
    >
      <Component />
    </div>
  </div>
</div>
`;

exports[`dashboard admitted 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design"
    >
      <Connect(Component)
        confirmedYet={false}
        deadline="2048-01-30T04:39:47.512Z"
      />
    </div>
  </div>
</div>
`;

exports[`dashboard incomplete after deadline 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design"
    >
      <span>
        Sorry, the application window has closed.
      </span>
    </div>
  </div>
</div>
`;

exports[`dashboard incomplete before deadline 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design"
    >
      <div>
        <div>
          You haven't submitted your 
          <Link
            replace={false}
            to="/application_info"
          >
            application
          </Link>
           yet. You have
        </div>
        <div
          className="days-left"
        >
          7627
        </div>
        <div>
          days to submit your application before the deadline:
          <br />
          <strong>
            November 18, 2019
          </strong>
          .
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`dashboard rejected 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design"
    >
      <Component />
    </div>
  </div>
</div>
`;

exports[`dashboard submitted 1`] = `
<div
  class="dashboard"
>
  <div
    class="stripe accent-blue"
  />
  <div
    class="stripe accent-orange bottom"
  />
  <div
    class="floating-illustration drone"
  >
    <img />
  </div>
  <div
    class="floating-illustration relax"
  >
    <img />
  </div>
  <div
    class="treehacks-dashboard-message-container"
  >
    <div
      class="dashboard-design"
    >
      <span>
        Your application has been received â€“ you are all good for now!
        <br />
        <br />
        We will email you when decisions are released and will handle any travel questions at that time. Thanks for applying :)
      </span>
    </div>
  </div>
</div>
`;

exports[`dashboard waitlisted 1`] = `
<div
  className="dashboard"
>
  <div
    className="stripe accent-blue"
  />
  <div
    className="stripe accent-orange bottom"
  />
  <div
    className="floating-illustration drone"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="floating-illustration relax"
  >
    <img
      src={[Function]}
    />
  </div>
  <div
    className="treehacks-dashboard-message-container"
  >
    <div
      className="dashboard-design"
    >
      <Component />
    </div>
  </div>
</div>
`;
